libs/langchain-mixedbread-ai/src/embeddings.ts

```diff
@@ -0,0 +1,166 @@
+
+type EmbeddingsRequestWithoutInput = Omit<MixedbreadAI.EmbeddingsRequest, "input">;
+
+/**
+ * Interface extending EmbeddingsParams with additional
+ * parameters specific to the MixedbreadAIEmbeddings class.
+ */
+export interface MixedbreadAIEmbeddingsParams extends EmbeddingsParams, Omit<EmbeddingsRequestWithoutInput, "model"> {
+  /**
+   * The model to use for generating embeddings.
+   * @default {"mixedbread-ai/mxbai-embed-large-v1"}
+   */
+  model?: string;
+
+  /**
+   * The maximum number of documents to embed in a single request.
+   * @default {128}
+   */
+  batchSize?: number;
+
+  /**
+   * The API key to use.
+   * @default {process.env.MXBAI_API_KEY}
+   */
+  apiKey?: string;
+
+  /**
+   * The base URL for the API.
+   */
+  baseUrl?: string;
+}
+
+/**
+ * Class for generating embeddings using the Mixedbread AI API.
+ *
+ * This class leverages the model "mixedbread-ai/mxbai-embed-large-v1" to generate
+ * embeddings for text documents. The embeddings can be used for various NLP tasks
+ * such as similarity comparison, clustering, or as features in machine learning models.
+ *
+ * @example
+ * const embeddings = new MixedbreadAIEmbeddings({ apiKey: 'your-api-key' });
+ * const texts = ["Baking bread is fun", "I love baking"];
+ * const result = await embeddings.embedDocuments(texts);
+ * console.log(result);
+ *
+ * @example
+ * const embeddings = new MixedbreadAIEmbeddings({
+ *  apiKey: 'your-api-key',
+ *  model: 'mixedbread-ai/mxbai-embed-large-v1',
+ *  encodingFormat: MixedbreadAI.EncodingFormat.Binary,
+ *  dimensions: 512,
+ *  normalized: true,
+ * });
+ * const texts = ["Baking bread is fun", "I love baking"];
+ * const result = await embeddings.embedDocuments(texts);
+ * console.log(result);
+ */
+export class MixedbreadAIEmbeddings extends Embeddings {
+  lc_secrets = {
+    apiKey: "MXBAI_API_KEY",
+  }
+
+  requestParams: EmbeddingsRequestWithoutInput;
+
+  batchSize: number;
+
+  private client: MixedbreadAIClient;
+
+  /**
+   * Constructor for MixedbreadAIEmbeddings.
+   * @param {Partial<MixedbreadAIEmbeddingsParams>} params - An optional object with properties to configure the instance.
+   * @throws {Error} If the API key is not provided or found in the environment variables.
+   * @throws {Error} If the batch size exceeds 256.
+   *
+   * @example
+   * const embeddings = new MixedbreadAIEmbeddings({
+   *     apiKey: 'your-api-key',
+   *     model: 'mixedbread-ai/mxbai-embed-large-v1',
+   *     batchSize: 64
+   * });
+   */
+  constructor(params?: Partial<MixedbreadAIEmbeddingsParams>) {
+    super({ maxConcurrency: 2, ...(params ?? {}) });
+
+    const apiKey = params?.apiKey ?? getEnvironmentVariable("MXBAI_API_KEY");
+    if (!apiKey) {
+      throw new Error("mixedbread ai API key not found. Either provide it in the constructor or set the 'MXBAI_API_KEY' environment variable.");
+    }
+    if (params?.batchSize && params?.batchSize > 256) {
```

Comment:
nit: Generally prefer setting this server side in case you change this in the future, but not blocking